### PR TITLE
Move linter code into top-level package

### PR DESCRIPTION
**Pull request by jnmugerwa**
Moved lint and warning code into its own top-level package (i.e. sibling of parser rather than child of parser).

### CODE REVIEW

File: linter/pom.xml [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        <dependency>
+            <groupId>com.facebook.presto</groupId>
+            <artifactId>presto-coresql-parser</artifactId>
+            <version>0.1-SNAPSHOT</version>
+            <scope>compile</scope>
+        </dependency>
+    </dependencies>
+    <build>
+        <plugins>
+            <plugin>
```
Comment (rongrong) [inline linter/pom.xml:45]: Do we need all these plugins here?

Comment (jnmugerwa) [reply linter/pom.xml:45]: Nope -- just removed all unused plugins and pushed common ones to the parent pom.

File: linter/pom.xml [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+                    <version>2.1.0</version>
+                    <configuration>
+                        <javaVersion>1.8</javaVersion>
+                        <ignorePackages>com.facebook.coresql.parser</ignorePackages>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
+</project>
```
Comment (rongrong) [inline linter/pom.xml:160]: nits: add a new line here

File: pom.xml [hunk 3]
```diff
@@ -91,4 +92,4 @@
             </plugin>
         </plugins>
     </build>
-</project>
+</project>
```
Comment (rongrong) [inline pom.xml:95]: nits: add a new line here.

**Review by rongrong — APPROVED** [review]

File: linter/src/main/java/com/facebook/coresql/linter/lint/LintingVisitor.java [hunk 4]
```diff
@@ -12,12 +12,12 @@
  * limitations under the License.
  */
 
-package com.facebook.coresql.lint;
+package com.facebook.coresql.linter.lint;
 
+import com.facebook.coresql.linter.warning.WarningCode;
```
Comment (rongrong) [inline linter/src/main/java/com/facebook/coresql/linter/lint/LintingVisitor.java:17]: I see in your other PR that you are adding warnings from rewriter. In that case should warning belong to linter?

Comment (jnmugerwa) [reply linter/src/main/java/com/facebook/coresql/linter/lint/LintingVisitor.java:17]: I intend this PR to contain the correct structure of all linter/warning code. After merging this branch, I'm going to merge it into the other branches (which may be confusing since -- like you mention -- there is a conflict at the moment).

Done

File: linter/pom.xml (hunk 2 above)
Comment (jnmugerwa) [reply linter/pom.xml:160]: Done

File: pom.xml (hunk 3 above)
Comment (jnmugerwa) [reply pom.xml:95]: Done

**Review by rongrong — CHANGES_REQUESTED** [review]

File: linter/pom.xml [hunk 5]
```diff
@@ -0,0 +1,182 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>com.facebook.presto</groupId>
+        <artifactId>presto-coresql</artifactId>
+        <version>0.1</version>
```
[the remark this answers is not in the document]
Comment (jnmugerwa) [reply linter/pom.xml]: Changed to most recent version -- 0.2-SNAPSHOT

Done

File: linter/pom.xml [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+        <dependency>
+            <groupId>org.testng</groupId>
+            <artifactId>testng</artifactId>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>com.facebook.presto</groupId>
+            <artifactId>presto-coresql-parser</artifactId>
+            <version>0.1</version>
```
[the remark this answers is not in the document]
Comment (jnmugerwa) [reply linter/pom.xml]: Changed all versions of dependencies to be consistent (i.e. all are now 0.2-SNAPSHOT)

File: parser/pom.xml [hunk 7]
```diff
@@ -1,36 +1,37 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
+
     <parent>
         <groupId>com.facebook.presto</groupId>
         <artifactId>presto-coresql</artifactId>
-        <version>0.1-SNAPSHOT</version>
+        <version>0.2-SNAPSHOT</version>
```
Comment (rongrong) [inline parser/pom.xml:8]: Why this change is here? The source code says 0.2. I don't quite understand this...

**Review by rongrong — APPROVED** [review]
